extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -0,0 +1,97 @@
+const { usersBucket, mpuBucketPrefix } = require('arsenal').constants;
+const QueuePopulatorExtension =
+    require('../../lib/queuePopulator/QueuePopulatorExtension');
+
+class LifecycleQueuePopulator extends QueuePopulatorExtension {
+
+    /**
+     * Pre-create the zookeeper path for bucket lifecycle nodes, if necessary.
+     * @param {Function} cb - The callback function.
+     * @return {undefined}
+     */
+    createZkPath(cb) {
+        const { zookeeperPath } = this.extConfig;
+        return this.zkClient.getData(zookeeperPath, err => {
+            if (err) {
+                if (err.name !== 'NO_NODE') {
+                    this.log.error('Could not get zookeeper node path', {
+                        method: 'LifecycleQueuePopulator.createZkPath',
+                        error: err,
+                    });
+                    return cb(err);
+                }
+                return this.zkClient.mkdirp(zookeeperPath, err => {
+                    if (err) {
+                        this.log.error('Could not create path in zookeeper', {
+                            method: 'LifecycleQueuePopulator.createZkPath',
+                            zookeeperPath,
+                            error: err,
+                        });
+                        return cb(err);
+                    }
+                    return cb();
+                });
+            }
+            return cb();
+        });
+    }
+
+    /**
+     * Create a new zookeeper node for any bucket that has a lifecycle
+     * configuration. Remove the node if the bucket is deleted or the
+     * configuration is deleted.
+     * @param {Object} value - The record log entry value from metadata.
+     * @return {undefined}
+     */
+    _updateZkBucketNode(value) {
+        const { zookeeperPath } = this.extConfig;
+        const path =
+            `${zookeeperPath}/data/buckets/${value.owner}:${value.name}`;
+        // Remove existing node if deleting the bucket or its configuration.
+        if (value.deleted || value.lifecycleConfiguration === null) {
+            return this.zkClient.remove(path, err => {
+                if (err && err.name !== 'NO_NODE') {
+                    this.log.error('Could not remove zookeeper node', {
+                        method: 'LifecycleQueuePopulator._updateZkBucketNode',
+                        zkPath: path,
+                        error: err,
+                    });
+                }
+                return undefined;
+            });
+        }
+        if (value.lifecycleConfiguration !== null) {
```

Comment:
This test can be removed (always true) because of the previous test at https://github.com/scality/backbeat/pull/172/files#diff-6c9b356900ebc28b298a5334d1b1b442R51

Reply:
Updated and squashed.